hudi-common/src/main/java/org/apache/hudi/avro/AvroLogicalTypeEnum.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.avro;
+
+public enum AvroLogicalTypeEnum {
```

Comment:
nit: Javadocs.  Does this work with all Avro spec versions?

Reply:
added comments

It is not a good idea to use the LogicalTypes because:
1. There is no enum/const string from avro side that give us an overview of all logical types that avro support by default. The LogicalTypes are java class, which are not a good choice to use class as parameters and pass it everywhere.
2. There are some logical types that the avro official doc claims have support (for example the uuid), while the avro lib of the same version does not have such logical type defined.
3. Logical types are intended for extension from client side. In the future if new customized logical type to be added, the enum approach is more friendly to be extended.

Reply:
It's still good to use contants from `LogicalTypes` here.

> There are some logical types that the avro official doc claims have support (for example the uuid), while the avro lib of the same version does not have such logical type defined.

Could you add some docs and link to the spec on what logical types are supported in which versions?  Hudi uses different versions of Avro depending on Spark and Flink versions.

Reply:
addressed offline, Ethan is on the same page that we create our own Enum as Avro does not have such thing published for consumption.

Reply:
Sg